stats/src/main/java/io/airlift/stats/JmxGcMonitor.java

```diff
@@ -140,6 +147,20 @@ private synchronized void onNotification(Notification notification)
                 long applicationRuntime = max(0, info.getStartTime() - lastGcEndTime);
                 lastGcEndTime = info.getEndTime();
 
+                if (info.getBeforeGcTotal().toBytes() / heapMemoryLimit > .8) {
+                    // Only warn of GC issues when the Heap is close to the limit
+                    double percentRecovered = 1 - info.getBeforeGcTotal().toBytes() / info.getAfterGcTotal().toBytes();
+                    if (percentRecovered < .1) {
+                        log.warn(
+                                "Major GC: application %sms, stopped %sms: %s -> %s (only recovered %s% of heap memory)",
+                                applicationRuntime,
+                                info.getDurationMs(),
+                                info.getBeforeGcTotal(),
+                                info.getAfterGcTotal(),
+                                percentRecovered);
```

Comment:
This is a fraction/ratio not a percentage, so need to multiply here

Reply:
good catch.